content/chapters/io/lecture/slides/devices.md

```diff
@@ -0,0 +1,172 @@
+## Devices
+
+---
+
+### I/O Devices
+
+- A hardware component that communicates with our application through bytes
+- Various devices fit this description:
+
+----
+
+![mouse keyboard](./media/dev-mouse-keyboard.png)
+
+----
+
+![hard disk](./media/dev-storage.png)
+
+----
+
+![sensors](./media/dev-sensors.png)
+
+----
+
+![devices](./media/devices.png)
+
+How does the OS operates them?
+
+---
+
+### Device Types
+
+| Char Devices          | Block Devices             |
+| :-------------------: | :-----------------------: |
+| read/write one byte   | read/write blocks of data |
+| slow                  | fast                      |
+| no search             | search                    |
+| no buffering          | buffering                 |
+
```

Comment:
I would give some `dd` examples that show how information is "stored" for block devices and any time you access it you get the same data. Unlike a char device such as `/dev/urandom`.

Reply:
I went with the first approach.